org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -153,17 +155,24 @@ public void initialized(InitializedParams params) {
 	 */
 	@Override
 	public CompletableFuture<Object> shutdown() {
-		logInfo(">> shutdown");
-		JavaLanguageServerPlugin.getLanguageServer().shutdown();
-		return CompletableFuture.completedFuture(new Object());
-	}
+    logInfo(">> shutdown");
+    return computeAsync((cc)->{
+      try {
+        ResourcesPlugin.getWorkspace().save(true, toMonitor(cc));
+      } catch (CoreException e) {
+        logException(e.getMessage(),e);
+      }
+      return new Object();
+    });
+  }
 
 	/* (non-Javadoc)
 	 * @see org.eclipse.lsp4j.services.LanguageServer#exit()
 	 */
 	@Override
 	public void exit() {
-		logInfo(">> exit");
+    logInfo(">> exit");
+    JavaLanguageServerPlugin.getLanguageServer().shutdown();
```

Comment:
so JDTLanguageServer.exit() calls LanguagServer.shutdown(), but JDTLanguageServer.shutdown() calls... nothing. It's a bit confusing

Reply:
JDTLanguageServer.shutdown()  saves the workspace.

Reply:
you should probably rename LanguageServer.shutdown() to LanguageServer.exit(), so it's consistent